sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultStep.java

```diff
@@ -139,6 +139,9 @@ public String getName() {
     @Override
     public Status getStatus() {
         synchronized (statusLock) {
+            if (status == Status.PENDING && getStrategy().isInterrupted()) {
+                return Status.WAITING;
```

Comment:
Well, but status does depend on whether or not the phase interrupted, thus this change. I'm just wondering if there's a point at which we can actively set `status` to `WAITING`, since I could see another method in the future accessing the member directly, bypassing this logic and getting the wrong answer.